pkg/metric/service/api/metric.proto

```diff
@@ -0,0 +1,130 @@
+/*
+ Copyright 2023 The Kapacity Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+syntax = "proto3";
+
+package service;
+option go_package = ".;api";
+
+message QueryLatestRequest {
+  MetricQuery query = 1;
+}
+
+message QueryRequest {
+  MetricQuery query = 1;
+  int64 start = 2;
+  int64 end = 3;
+  int64 step = 4;
+}
+
+message QueryLatestResponse {
+  repeated Sample samples = 1;
+}
+
+message QueryResponse {
+  repeated Series series = 1;
+}
+
+message Sample {
+  Point point = 1;
+  map<string, string> labels = 2;
+  int64 window = 3;
+}
+
+message MetricQuery {
+  MetricQueryType type = 1;
+  oneof resourceQuery {
+    PodResourceQuery  podResource = 2;
+    ContainerResourceQuery containerResource = 3;
+    WorkloadResourceQuery workloadResource = 4;
+    WorkloadContainerResourceQuery workloadContainerResource = 5;
+    ObjectQuery object = 6;
+    ExternalQuery external = 7;
+  }
+}
+
+enum MetricQueryType {
+  PodResource = 0;
```

Comment:
According to [Protobuf Style Guide](https://protobuf.dev/programming-guides/style/#enums), all enum value names should use CAPITALS_WITH_UNDERSCORES, e.g. `POD_RESOURCE`